Source/PeleLMeX_Forces.cpp

```diff
@@ -171,3 +171,70 @@ PeleLM::getVelForces(
         gp0, dV_control, dx, vel, rho, rhoY, rhoh, temp, extMom, extRho, force);
     });
 }
+
+void
+PeleLM::addSpark(const int lev, const TimeStamp& a_timestamp)
+{
+
+  const Real* probLo = geom[lev].ProbLo();
+  auto const dx = geom[lev].CellSizeArray();
+  bool verb = m_spark_verbose > 1 && lev == 0 && a_timestamp == AmrOldTime;
+  for (int n = 0; n < m_n_sparks; n++) {
+    IntVect spark_idx;
+    Real time = getTime(lev, a_timestamp);
+    if (
+      time < m_spark_time[n] || time > m_spark_time[n] + m_spark_duration[n]) {
+      if (verb) {
+        Print() << m_spark[n] << " not active" << std::endl;
+      }
+      continue;
+    }
+    if (verb) {
+      Print() << m_spark[n] << " active" << std::endl;
+    }
+    for (int d = 0; d < AMREX_SPACEDIM; d++) {
+      spark_idx[d] = (int)((m_spark_location[n][d] - probLo[d]) / dx[d]);
+    }
+    Box domainBox = geom[lev].Domain();
+    // just a check
+    if (!domainBox.contains(spark_idx)) {
+      Warning(m_spark[n] + " not in domain!");
+      continue;
+    }
+    auto* ldata_p = getLevelDataPtr(lev, a_timestamp);
+    auto eos = pele::physics::PhysicsType::eos();
+    for (MFIter mfi(*(m_extSource[lev]), TilingIfNotGPU()); mfi.isValid();
```

Comment:
Add OMP pragma similar to other ParallelFors in this file:
```
#ifdef AMREX_USE_OMP
#pragma omp parallel if (Gpu::notInLaunchRegion())
#endif
```

Or, the better option would be to convert this into the newer version of ParallelFor that operates directly on a MultiFab without MFIter (see, e.g, https://github.com/AMReX-Combustion/PeleLMeX/blob/8dccce90dfba3a1000a1c1602e701a06a0d09b7b/Source/PeleLMeX_Advection.cpp#L749, and note the `Gpu::StreamSynchronize` at the end of the loop)

Reply:
I've made this change - I assume there's a lot of functions lying around that need this change...

Reply:
Yeah there probably are. The multifab level ParFor can be a bit more efficient on GPU and simplifies the code (not needing MFiter and OMP pragmas), but it isn't a big enough deal to be a priority to go back and switch the old ones. Lucas switched a bunch of them over quite a while ago in potentially higher impact places.